src/ImageSharp.Web/Commands/CommandParser.cs

```diff
@@ -61,99 +37,38 @@ private CommandParser()
-
-            this.AddConverter(typeof(double), typeof(SimpleCommandConverter<double>));
-            this.AddConverter(typeof(string), typeof(SimpleCommandConverter<string>));
-
-            this.AddConverter(typeof(bool), typeof(SimpleCommandConverter<bool>));
-        }
-
-        /// <summary>
-        /// Adds a selection of default list type converters.
-        /// </summary>
-        private void AddListConverters()
-        {
-            this.AddConverter(typeof(List<sbyte>), typeof(ListConverter<sbyte>));
-            this.AddConverter(typeof(List<byte>), typeof(ListConverter<byte>));
-
-            this.AddConverter(typeof(List<short>), typeof(ListConverter<short>));
-            this.AddConverter(typeof(List<ushort>), typeof(ListConverter<ushort>));
-
-            this.AddConverter(typeof(List<int>), typeof(ListConverter<int>));
-            this.AddConverter(typeof(List<uint>), typeof(ListConverter<uint>));
-
-            this.AddConverter(typeof(List<long>), typeof(ListConverter<long>));
-            this.AddConverter(typeof(List<ulong>), typeof(ListConverter<ulong>));
-
-            this.AddConverter(typeof(List<decimal>), typeof(ListConverter<decimal>));
-            this.AddConverter(typeof(List<float>), typeof(ListConverter<float>));
-            this.AddConverter(typeof(List<double>), typeof(ListConverter<double>));
-
-            this.AddConverter(typeof(List<string>), typeof(ListConverter<string>));
-        }
+            => (T)this.ParseValue(typeof(T), value, culture);
 
         /// <summary>
-        /// Adds a selection of default array type converters.
+        /// Parses the given string value converting it to the given type.
         /// </summary>
-        private void AddArrayConverters()
+        /// <param name="type"> The type to convert the string to.</param>
+        /// <param name="value">The string value to parse.</param>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use as the current culture.</param>
+        /// <returns>The converted instance or the default value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public object ParseValue(Type type, string value, CultureInfo culture)
         {
-            this.AddConverter(typeof(sbyte[]), typeof(ArrayConverter<sbyte>));
-            this.AddConverter(typeof(byte[]), typeof(ArrayConverter<byte>));
-
-            this.AddConverter(typeof(short[]), typeof(ArrayConverter<short>));
-            this.AddConverter(typeof(ushort[]), typeof(ArrayConverter<ushort>));
+            DebugGuard.NotNull(type, nameof(type));
+            DebugGuard.NotNull(culture, nameof(culture));
 
-            this.AddConverter(typeof(int[]), typeof(ArrayConverter<int>));
-            this.AddConverter(typeof(uint[]), typeof(ArrayConverter<uint>));
+            // This allows us to reuse the same converter for infinite enum types.
+            Type matchType = type.IsEnum ? typeof(Enum) : type;
```

Comment:
This does mean that users can't register custom Converters for specific enums however... should this be a fall back instead of a force change? i.e. looks for specific converter first, then use the generic Enum converter if one can't be found.

Reply:
Yep, wasn’t something I considered but can easily workaround. Will update 👍

Reply:
Done